plugin-server/src/utils/db/db.ts

```diff
@@ -673,109 +654,70 @@ export class DB {
-        this.promiseManager.trackPromise(this.updateGroupDataCache(teamId, result))
-        return result
-    }
+            this.statsd?.increment('group_info_cache.miss')
 
-    public async fetchGroupColumnsValues(teamId: number, groups: GroupIdentifier[]): Promise<Record<string, string>> {
-        if (groups.length === 0) {
-            return {}
-        }
-        const cachedResults = await Promise.all(
-            groups.map((groupIdentifier) => this.getGroupDataCache(teamId, groupIdentifier))
-        )
+            // If we didn't find cached data, lookup the group from Postgres
+            const storedGroupData = await this.fetchGroup(teamId, groupTypeIndex as GroupTypeIndex, groupKey)
 
-        let columns: Record<string, string> = this.groupCacheDataToColumns(cachedResults)
-        const groupsWithoutCachedData = cachedResults
-            .filter((groupData) => !groupData.cached)
-            .map((groupData) => groupData.identifier)
+            if (storedGroupData) {
+                groupColumns[propertiesColumnName] = JSON.stringify(storedGroupData.group_properties)
 
-        if (groupsWithoutCachedData.length > 0) {
-            const fetchedResults = await this.fetchGroupDataAndUpdateCache(teamId, groupsWithoutCachedData)
-            columns = { ...columns, ...this.groupCacheDataToColumns(fetchedResults) }
-        }
-        return columns
-    }
+                const createdAt = castTimestampOrNow(storedGroupData.created_at)
+                groupColumns[createdAtColumnName] = createdAt
 
-    private groupCacheDataToColumns(groupCacheData: GroupCacheData[]): Record<string, string> {
-        const result: Record<string, string> = {}
-        for (const group of groupCacheData) {
-            if (group.data) {
-                result[`group${group.identifier.index}_properties`] = JSON.stringify(group.data.properties)
-                result[`group${group.identifier.index}_created_at`] = castTimestampOrNow(
-                    group.data.created_at,
-                    TimestampFormat.ClickHouse
-                )
+                // We found data in Postgres, so update the cache
+                // We also don't want to throw here, worst case is we'll have to fetch from Postgres again next time
+                try {
+                    await this.updateGroupCache(teamId, groupTypeIndex, groupKey, {
+                        properties: storedGroupData.group_properties,
+                        created_at: createdAt,
+                    })
+                } catch (error) {
+                    captureException(error)
+                }
+            } else {
+                // We couldn't find the data from the cache nor Postgres, so record this in a metric and in Sentry
+                this.statsd?.increment('groups_data_missing_entirely')
+                captureException(new Error('Missing groups data entirely'), { extra: { groupCacheKey } })
+
+                groupColumns[propertiesColumnName] = '{}'
```

Comment:
what if we added a property that's internal telling us we hit this, so it will be easier for us to distinguish these cases from other empty ones?

Reply:
I've thought about these things before, but feels... hacky